pkg/server/share_manager.go

```diff
@@ -210,6 +240,96 @@ func (m *ShareManager) resizeVolume(devicePath, mountPath string) error {
 	return nil
 }
 
+func (m *ShareManager) makeLeaseClient() error {
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create InClusterConfig")
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create kubernetes client set")
+		return err
+	}
+
+	// Use the clientset to get the node name of the share-manager pod
+	// and store for use by takeLease().
+	podName := shareManagerPrefix + m.volume.Name
+	pod, err := clientset.CoreV1().Pods(namespace).Get(m.context, podName, metav1.GetOptions{})
+	if err == nil && pod != nil {
+		m.leaseHolder = pod.Spec.NodeName
+	} else {
+		m.leaseHolder = podName
+	}
+
+	m.leaseClient = clientset.CoordinationV1()
+	return nil
+}
+
+func (m *ShareManager) getLease() error {
+	if m.leaseClient == nil {
+		return fmt.Errorf("kubernetes API client is unset")
```

Comment:
This error does not make sense IMO. It seems it should have been an unrecoverable error to fail to get the lease client initially (in which case, the pod should have crashed out). If we want to allow the share manager pod to continue running without an initialized client, we should provide a path that doesn't involve continually logging the same unrecoverable error message.

Reply:
As above, it will log only once.  But I agree, getLease and takeLease can probably be collapsed into a single function and simplify things.